src/module/item/physical/document.ts

```diff
@@ -575,6 +577,33 @@ abstract class PhysicalItemPF2e<TParent extends ActorPF2e | null = ActorPF2e | n
             equipped["-=inSlot"] = null;
         }
 
+        // Special handling for ephemeral hit point changes: if the max HP would be adjusted when changing this item's
+        // carry type, proportionally adjust the current HP as well.
+        type MaybeItemAlteration = { key: string; property?: unknown; itemType?: unknown };
+        const hasHpChangeRules =
+            this.actor?.rules.some(
+                (r: MaybeItemAlteration) =>
+                    r.key === "ItemAlteration" && r.property === "hp-max" && r.itemType === this.type
+            ) ?? false;
+        if (hasHpChangeRules && changed.system?.equipped?.carryType) {
+            // Simulate the update to determine whether max hit points will change
+            const postUpdateHPMax = ((): number => {
+                const actorSource = this.actor!.toObject();
+                const itemSource = actorSource.items.find((i) => i._id === this.id) ?? { system: {} };
+                itemSource.system = mergeObject(itemSource.system, deepClone(changed.system));
+                const actorClone = new ActorProxyPF2e(actorSource);
+                const itemClone = actorClone.items.get(this.id, { strict: true });
+                return itemClone.isOfType("armor") ? itemClone.system.hp.max : this.system.hp.max;
+            })();
+            if (postUpdateHPMax !== this.system.hp.max) {
```

Comment:
```suggestion

            // Apply the value change, and avoid "broken" changing by rounding directionally
            if (postUpdateHPMax !== this.system.hp.max) {
```

Wasn't obvious to me that this is trying to make sure that broken is the same with the floorOrCeil ternary, and I only realized that because I remember you brought it up in the discord. The if statement might benefit from a comment explaining the intent here.